### PR TITLE
Sanitize context in generated Javadocs for version catalogs

### LINKED ISSUE #19752: VERSION_CATALOGS feature fails if any dir in project path starts with character "u": "illegal unicode escape"
```
FAILURE: Build failed with an exception.
* What went wrong:
org.gradle.api.internal.catalog.GeneratedClassCompilationException: Unable to compile generated sources:
  - File LibrariesForUnity.java, line: 31, illegal unicode escape
  - File LibrariesForUnity.java, line: 53, illegal unicode escape
> Unable to compile generated sources:
    - File LibrariesForUnity.java, line: 31, illegal unicode escape
    - File LibrariesForUnity.java, line: 53, illegal unicode escape
```

This is because "\u" is treated as a unicode escape even in comments:
`LibrariesForUnity.java` generated by gradle - note `\u` in the javadoc:
```java
...
        /**
         * Creates a dependency provider for platform (com.xxx:unity-xxx)
         * This dependency was declared in settings file '..\unityxxx\settings.gradle'
         */
        public Provider<MinimalExternalModuleDependency> getXxxxxx() { return create("xxxxxx"); }
...
```

This is happening when importing such project to IntelliJ IDEA version 2021.3.2 RC Build #IU-213.6777.38.

`unityxxx/settings.gradle`:
```groovy

include 'aaa', 'bbb'

enableFeaturePreview('VERSION_CATALOGS')
dependencyResolutionManagement {
    versionCatalogs {
        unity {
            version('unity', '3+')
            alias('p').to('com.xxx.xxx', 'unity-p').versionRef('unity')
        }
    }
}
```

`unityxxx/bbb/build.gradle`:
```groovy

dependencies {
    implementation unity.p
}
```

### PR DESCRIPTION
**Pull request by alllex**
Fixes #19752

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test this

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java [hunk 1]
```diff
@@ -491,6 +491,14 @@ private void writePlugin(String alias, String id, @Nullable String context, bool
         writeLn();
     }
 
+    /**
+     * Java compiler would fail to compile sources that have illegal unicode escape characters in the comments.
+     * Such characters could be accidentally introduced by an unescaped backslash.
+     */
+    private static String escapeBackslashForJavaComment(String s) {
+        return s.replace("\\", "\\\\");
```
Comment (ljacomet) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java:499]: I am not too fond of this solution, the end result in the javadoc is something like:
`C:\\Users\\user\\Documents\\ultimate plugin` where all `\` a doubled.

Why don't we address the exact problem and replace `\u` with `\u005cu` which both an IDE (when viewing Javadoc) and the generated Javadoc would render as `\u`.

**Review by ljacomet — CHANGES_REQUESTED** [review]
See below for an alternative solution

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java (hunk 1 above)
Comment (alllex) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java:499]: This is a nice solution, although at the same time if the javadoc is not rendered, then the path looks even more unreadble/broken: `C:\Users\u005cuser\Documents\u005cultimate plugin`

**Comment by alllex** [comment]
@bot-gradle test qf

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java (hunk 1 above)
Comment (alllex) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java]: It was decided to go with @ljacomet 's solution 👍

**Review by ljacomet — APPROVED** [review]
LGTM! :shipit:

**Comment by alllex** [comment]
@bot-gradle test and merge

**Comment by alllex** [comment]
@bot-gradle test and merge

**Pull request merged**